### PR TITLE
Better write array with benchmark

### PR DESCRIPTION
**Pull request by vahid-sohrabloo**
For better performance, I add a write length function. For check it I also provide a benchmark.
this is a result of my benchmark
```
Benchmark_BlockWriteArrayWithLen4-4   	32752696	        45.5 ns/op	      49 B/op	       0 allocs/op
Benchmark_BlockWriteArray4-4          	 4509878	       264 ns/op	     113 B/op	       4 allocs/op
```

### CODE REVIEW

File: lib/data/block.go [hunk 1]
```diff
@@ -95,20 +95,24 @@ func (block *Block) Read(serverInfo *ServerInfo, decoder *binary.Decoder) (err e
 	return nil
 }
 
+func (block *Block) writeLen(l, num, level int) {
```
Comment (cw9) [inline lib/data/block.go]: use a more intuitive name instead of `l`?

File: lib/data/block_benchmark_test.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	block.Columns = append(block.Columns, c)
+	block.Reserve()
+	b.ReportAllocs()
+
+	for i := 0; i < b.N; i++ {
+		block.WriteArrayLen(4, 0, 1)
+		for i := 0; i > 4; i++ {
+			block.WriteInt32(0, 1)
+		}
+	}
```
Comment (cw9) [inline lib/data/block_benchmark_test.go]: This only works for one level arrays? Does it work for Nested Arrays as well?

File: lib/data/block.go (hunk 1 above)
Comment (vahid-sohrabloo) [reply lib/data/block.go]: I changed it

File: lib/data/block_benchmark_test.go (hunk 2 above)
Comment (vahid-sohrabloo) [reply lib/data/block_benchmark_test.go]: I add another benchmark

**Pull request closed without merging**